server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailHeader.scala

```diff
@@ -75,12 +76,35 @@ object GroupedAddressesHeaderValue extends EmailHeaderValue {
   }
 }
 
+object MessageIdsHeaderValue {
+  def from(field: Field): MessageIdsHeaderValue = {
+    val messageIds: List[HeaderMessageId] = format(field.getBody)
+      .split(" ")
+      .flatMap(body => {
+        if(Pattern.compile("^<.*@.*>$").matcher(body).matches()) {
+          scala.Right(body.replaceAll("[<,>]", ""))
+        } else {
+          Left()
+        }
+      }.toOption)
+      .toList
+      .map(HeaderMessageId(_))
+
+      MessageIdsHeaderValue(Option(messageIds).filter(_.nonEmpty))
+  }
+
+  def format(body: String): String = {
+    MimeUtil.unfold(body).trim().replace("\\s+", " ")
+  }
+}
+
 case class EmailHeaderName(value: String) extends AnyVal
 
 sealed trait EmailHeaderValue
 case class RawHeaderValue(value: String) extends EmailHeaderValue
 case class TextHeaderValue(value: String) extends EmailHeaderValue
 case class AddressesHeaderValue(value: List[EmailAddress]) extends EmailHeaderValue
 case class GroupedAddressesHeaderValue(value: List[EmailAddressGroup]) extends EmailHeaderValue
+case class MessageIdsHeaderValue(value: Option[List[HeaderMessageId]]) extends EmailHeaderValue
```

Comment:
Use MessageIdsHeaderValue as email `messageId`, `inReplyTo` and `references` fields.